x/accounts/vesting/continuous_vesting_account.go

```diff
@@ -0,0 +1,173 @@
+package vesting
+
+import (
+	"context"
+	"time"
+
+	"cosmossdk.io/collections"
+	errorsmod "cosmossdk.io/errors"
+	"cosmossdk.io/math"
+	"cosmossdk.io/x/accounts/accountstd"
+	account_abstractionv1 "cosmossdk.io/x/accounts/interfaces/account_abstraction/v1"
+	vestingtypes "cosmossdk.io/x/accounts/vesting/types"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+// Compile-time type assertions
+var (
+	_ accountstd.Interface = (*ContinuousVestingAccount)(nil)
+)
+
+// NewContinuousVestingAccount creates a new ContinuousVestingAccount object.
+func NewContinuousVestingAccount(d accountstd.Dependencies) (*ContinuousVestingAccount, error) {
+	baseVestingAccount := NewBaseVesting(d)
```

Comment:
Where is the implementation for `NewBaseVesting`. Does it implicitly create a base account? If so, I believe we want to avoid that.

Reply:
ah no, it called baseVesting just because it create the share fields of all vesting type, it does not create an auth base account

Reply:
i think i will rename the variable name to avoid confusion